machine/machine_core/ssh_connection.py

```diff
@@ -306,70 +301,25 @@ def execute(self, command=None, script=None, input=None, environment={},
 
         additional_ssh_params += self.__execution_opts(direct=direct)
 
-        if command:
-            if getattr(command, "strip", None):  # Is this a string?
-                cmd += [command]
-                if not quiet:
-                    self.message("+", command)
-            else:
-                # use shlex.join() once Python 3.8 is available everywhere
-                cmd.append(' '.join(shlex.quote(arg) for arg in command))
-                if not quiet:
-                    self.message("+", *command)
+        if getattr(command, "strip", None):  # Is this a string?
```

Comment:
I could have sworn we changed this to `isinstance()`...

Reply:
No, I backed out that commit from PR #3706 as you weren't happy about the aliasing, and I wasn't happy with any other solution.